services/frontend-service/src/ui/Batch.tsx

```diff
@@ -0,0 +1,144 @@
+import { BatchAction } from '../api/api';
+import { useUnaryCallback } from './Api';
+import * as React from 'react';
+import { Button, Dialog, DialogTitle, IconButton, Typography, Snackbar } from '@material-ui/core';
+import { useCallback } from 'react';
+import { Close } from '@material-ui/icons';
+
+export const useBatch = (act: BatchAction, fin?: () => void) =>
+    useUnaryCallback(
+        React.useCallback(
+            (api) =>
+                api
+                    .batchService()
+                    .ProcessBatch({
+                        actions: [act],
+                    })
+                    .finally(fin),
+            [act, fin]
+        )
+    );
+
+export interface ConfirmationDialogProviderProps {
+    children: React.ReactElement;
+    action: BatchAction;
+    fin?: () => void;
+}
+
+export const ConfirmationDialogProvider = (props: ConfirmationDialogProviderProps) => {
+    const { action, fin } = props;
+    const [openNotify, setOpenNotify] = React.useState(false);
+    const [openDialog, setOpenDialog] = React.useState(false);
+
+    const openNotification = useCallback(() => {
+        setOpenNotify(true);
+    }, [setOpenNotify]);
+
+    const closeNotification = useCallback(
+        (event: React.SyntheticEvent | React.MouseEvent, reason?: string) => {
+            if (reason === 'clickaway') {
+                return;
+            }
+            setOpenNotify(false);
+        },
+        [setOpenNotify]
+    );
+
+    const handleOpen = useCallback(() => {
+        setOpenDialog(true);
+        setOpenNotify(false);
+    }, [setOpenDialog, setOpenNotify]);
+
+    const handleClose = useCallback(() => {
+        setOpenDialog(false);
+    }, [setOpenDialog]);
+
+    const closeWhenDone = useCallback(() => {
+        openNotification();
+        if (fin) fin();
+        handleClose();
+    }, [fin, handleClose, openNotification]);
+
+    const [doAction, doActionState] = useBatch(action, closeWhenDone);
+
+    const closeIcon = (
+        <IconButton size="small" aria-label="close" color="secondary" onClick={closeNotification}>
+            <Close fontSize="small" />
+        </IconButton>
+    );
+
+    let title = '';
+    let notMessageSuccess = '';
+    let notMessageFail = '';
+
+    switch (action.action?.$case) {
+        case 'deploy':
+            title = 'Are you sure you want to deploy this version?';
+            notMessageSuccess =
+                'Version ' +
+                action.action?.deploy.version +
+                ' was successfully deployed to ' +
+                action.action?.deploy.environment;
+            notMessageFail = 'Deployment failed';
+            break;
```

Comment:
not critical, but I would suggest to move this whole switch into it's own function.
If you return an object with three fields, it would not require so many statements (just expressions).
```
function getMessagesForState(state) {
 case 'deploy':
   return { 
   title: ...
   message: ...
   }
```